Cabal/src/Distribution/Simple/Glob.hs

```diff
@@ -213,7 +218,14 @@ matchDirFileGlobWithDie verbosity rip version mbWorkDir symPath =
 
 -- ** Parsing globs in a cabal package
 
-parseFileGlob :: CabalSpecVersion -> FilePath -> Either GlobSyntaxError Glob
+-- | This function returns a nested 'Either' - If the resulting 'Glob'
+-- would be a completely literal file-path, then we return a @'Right'
+-- ('Left' filepath)@. This allows code to avoid going through an expensive
+-- 'Glob' match when a much faster filepath operation is available.
+parseFileGlob :: CabalSpecVersion -> FilePath -> Either GlobSyntaxError (Either FilePath Glob)
+parseFileGlob _version filepath
+  | all (`notElem` filepath) "*{}," =
+      Right (Left filepath)
```

Comment:
This is the core change in this PR. If we don't actually use any glob-like characters, it's not a glob, so we can skip the slow glob treatment.